### PR TITLE
Fix leaking of adb processes

### PR DESCRIPTION
**Pull request by rosen-vladimirov**
When we print device logs, we execute `adb -s <device identifier> logcat`.
In case the device is not unlocked, we receive `undefined` for deviceIdentifier and the command leaks adb processes.
It looks like executing `adb -s <invalid id> logcat` is not closed even after our process exits.
Fix this by checking the passed value. Also change the return type of `LogcatHelper`'s `start` method to void
as noone is using the returned child_process.

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: mobile/android/logcat-helper.ts [hunk 1]
```diff
@@ -10,28 +10,28 @@ export class LogcatHelper implements Mobile.ILogcatHelper {
...
 
-	public start(deviceIdentifier: string): any {
-		let adbPath = this.$staticConfig.getAdbFilePath().wait();
-		this.$childProcess.exec(util.format("%s -s %s logcat -c", adbPath, deviceIdentifier)).wait(); // remove cached logs
-		let adbLogcat = this.$childProcess.spawn(adbPath, ["-s", deviceIdentifier, "logcat"]);
-		let lineStream = byline(adbLogcat.stdout);
+	public start(deviceIdentifier: string): void {
+		if(deviceIdentifier) {
+			let adbPath = this.$staticConfig.getAdbFilePath().wait();
+			this.$childProcess.exec(util.format("%s -s %s logcat -c", adbPath, deviceIdentifier)).wait(); // remove cached logs
```
Comment (teobugslayer) [inline mobile/android/logcat-helper.ts]: consider spawning instead of execing

Comment (rosen-vladimirov) [inline mobile/android/logcat-helper.ts]: this will be handled in a separate PR (related to all changes I'm making in NativeScript CLI)

**Comment by Mitko-Kerezov** [comment]
:+1:
